index.src.html

```diff
@@ -2211,90 +2260,99 @@
-    [=a promise rejected with=] {{InvalidStateError}} {{DOMException}}.
-2. Let |p| be a new Promise.
-3. In parallel, resolve |p| with the result of running the <a>Clone Frame</a>
-    algorithm with <a>this</a>.
-4. Return |p|.
+: <dfn method for=VideoFrame>close()</dfn>
+:: Clears all state and releases the reference to the [=frame resource=].
+    Close is final.
 
-<dfn method for=VideoFrame>createImageBitmap(options)</dfn>
-Creates an ImageBitmap from this {{VideoFrame}}.
+    When invoked, run these steps:
+    1. Assign `null` to {{VideoFrame/[[resource reference]]}}.
+    2. Assign `true` to {{VideoFrame/[[detached]]}}.
+    3. Assign `""` to {{VideoFrame/format}}.
+    4. Assign `null` to {{VideoFrame/planes}}.
+    5. Assign `0` to {{VideoFrame/codedWidth}}, {{VideoFrame/codedHeight}},
+        {{VideoFrame/cropLeft}}, {{VideoFrame/cropTop}},
+        {{VideoFrame/cropWidth}}, {{VideoFrame/cropHeight}},
+        {{VideoFrame/displayWidth}}, and {{VideoFrame/displayHeight}}.
+    6. Assign `null` to {{VideoFrame/duration}} and {{VideoFrame/timestamp}}.
+
+: <dfn method for=VideoFrame>createImageBitmap(options)</dfn>
+:: Creates an ImageBitmap from this {{VideoFrame}}.
 
-When invoked, run these steps:
-1. Let |p| be a new Promise.
-2. If either |options|'s {{ImageBitmapOptions/resizeWidth}} or
-    {{ImageBitmap/resizeHeight}} is present and is 0, then return |p| rejected
-    with an {{InvalidStateError}} {{DOMException}}.
-3. If the <a>this'</a> {{VideoFrame/[[detached]]}} internal slot is set to
-    `true`, then return |p| rejected with an {{InvalidStateError}}
-    {{DOMException}}.
-4. Let |imageBitmap| be a new {{ImageBitmap}} object.
-5. Set |imageBitmap|'s bitmap data to a copy of the {{VideoFrame}} pixel data,
-    at the frame's intrinsic width and intrinsic height (`i.e`., after any
-    aspect-ratio correction has been applied), cropped to the source rectangle
-    with formatting.
-6. If the origin of |imageBitmap|'s image is not same origin with entry settings
-    object's origin, then set the origin-clean flag of |imageBitmap|'s bitmap to
-    `false`.
-7. Run this step in parallel:
-  1. Resolve p with imageBitmap.
+    When invoked, run these steps:
+    1. Let |p| be a new Promise.
+    2. If either |options|'s {{ImageBitmapOptions/resizeWidth}} or
+        {{ImageBitmap/resizeHeight}} is present and is 0, then return |p|
+        rejected with an {{InvalidStateError}} {{DOMException}}.
+    3. If the <a>this'</a> {{VideoFrame/[[detached]]}} internal slot is set to
+        `true`, then return |p| rejected with an {{InvalidStateError}}
+        {{DOMException}}.
+    4. Let |imageBitmap| be a new {{ImageBitmap}} object.
+    5. Set |imageBitmap|'s bitmap data to a copy of the {{VideoFrame}} pixel
+        data, at the frame's intrinsic width and intrinsic height (`i.e`.,
+        after any aspect-ratio correction has been applied),
+        [=ImageBitmap/cropped to the source rectangle with formatting=].
```

Comment:
What about all the other options? In particular, we really want to make sure colors are correct here, we can't just copy the bytes I think.

Reply:
Discussed in editors call:  I borrowed the language here from createImageBitmap(). It seems the options are applied when the ImageBitamp is created, but baked into the the bitmap data from that point forward. This particular step in our algorithm is still useful, as it describes that we are cropping the full image (from (0,0) -> (width, height).